cypress/integration/05_collection_spec.js

```diff
@@ -1,164 +1,332 @@
 /// <reference types="Cypress" />
+import { FSApi } from '../../fusion-studio-extension/src/common/api';
 
 context('Collection Operations', () => {
-  let fetchSpy;
   describe('working with tree view', () => {
+    const connection = {
+      server: Cypress.env('API_HOST'),
+      username: 'admin',
+      password: '',
+    };
     before(() => {
+      // prepare collections used in the test
+      new Cypress.Promise(async resolve => {
+        await FSApi.remove(connection, '/db/test', true).catch(e => { });
+        await FSApi.newCollection(connection, '/db/test');
+        await FSApi.newCollection(connection, '/db/test/col1');
+        await FSApi.save(connection, '/db/test/col1/test.txt', 'test text file');
+        resolve();
+      })
       cy.connect()
       cy.visit('/');
     })
     beforeEach(() => {
-      cy.window().then(win => fetchSpy = cy.spy(win, 'fetch').as('fetch'));
+      cy.window().then(win => cy.spy(win, 'fetch').as('fetch'));
+    })
+    after(() => {
+      // delete the test colelction
+      new Cypress.Promise(resolve => FSApi.remove(connection, '/db/test', true).then(resolve).catch(resolve))
```

Comment:
The before and after hooks of the document spec are better, please match them here.